### PR TITLE
Integrate BanyanDB as storage solution

### PR DESCRIPTION
**Pull request by ButterBright**
Integrate BanyanDB as storage solution.

### CODE REVIEW

File: chart/banyandb/templates/deployment.yaml
Comment (hanahmily) [inline chart/banyandb/templates/deployment.yaml:19]: This should be a statefulset.

File: CHANGES.md
Comment (kezhenxu94) [inline CHANGES.md:17]: 4.5.0 is released, please don't change the changelog here, add a new section for 4.6.0

**Comment by ButterBright** [comment]
Okay, I got it.

File: test/e2e/e2e.yaml
Comment (innerpeacez) [inline test/e2e/e2e.yaml:57]: Both ES and BanyanDB should be covered by e2e tests.

File: chart/banyandb/Chart.yaml
Comment (kezhenxu94) [inline chart/banyandb/Chart.yaml:39]: Let's clean up the comments in this file, most of the comments are not necessary

File: chart/skywalking/Chart.yaml
Comment (kezhenxu94) [inline chart/skywalking/Chart.yaml:44]: I think this doesn't work (?) when we release the skywalking Chart to Docker hub, using this installation method https://github.com/apache/skywalking-kubernetes#install-released-version-using-docker-helm-repository--430

Comment (ButterBright) [reply chart/skywalking/Chart.yaml:44]: What should I change the repo to? Something like oci://registry-1.docker.io/apache/skywalking-banyandb-helm?

**Review by wu-sheng — CHANGES_REQUESTED** [review]
Hold due to new repo should be created.

Mail list, https://lists.apache.org/thread/l6c9s5t2opljbrs5fsd6xs42nh4lh7bn

@ButterBright Please reach @hanahmily for more context. And most of your PR should be reusable.

**Comment by ButterBright** [comment]
I see.

**Review by hanahmily — APPROVED** [review]
LGTM

File: chart/skywalking/Chart.yaml
Comment (wu-sheng) [inline chart/skywalking/Chart.yaml:43]: I don't think we could use this as the final release. We should request for docker hub repository, and do a release for banyandb helm. @hanahmily

Comment (kezhenxu94) [reply chart/skywalking/Chart.yaml:43]: Because we deploy skywalking chart snapshot to ghcr so this makes sense in main branch but have to remember to change to a released one for banyandb chart before releasing.  This also implies that a skywalking chart release depends on a banyandb release.

Comment (hanahmily) [reply chart/skywalking/Chart.yaml:43]: Makes sense to me. We will release the banyandb helm once the cluster mode is supported later. @ButterBright please make a note of this.

Comment (wu-sheng) [reply chart/skywalking/Chart.yaml:43]: > Because we deploy skywalking chart snapshot to ghcr so this makes sense in main branch but have to remember to change to a released one for banyandb chart before releasing. This also implies that a skywalking chart release depends on a banyandb release.

For me, the concern is, that we don't release `helm` repo in high frequency, so, most likely we definitely will lose track and release the next helm with this snapshot repository.

This PR is never urgent, so, if you feel the BanyanDB helm repo is ready to release, please go ahead to do so. I believe no matter banyandb helm repo is relying on BanyanDB server 0.5 release or not, it is close to be ready.

Comment (ButterBright) [reply chart/skywalking/Chart.yaml:43]: > Makes sense to me. We will release the banyandb helm once the cluster mode is supported later. @ButterBright please make a note of this.

Okay.

Comment (kezhenxu94) [reply chart/skywalking/Chart.yaml:43]: > > Because we deploy skywalking chart snapshot to ghcr so this makes sense in main branch but have to remember to change to a released one for banyandb chart before releasing. This also implies that a skywalking chart release depends on a banyandb release.
> 
> For me, the concern is, that we don't release `helm` repo in high frequency, so, most likely we definitely will lose track and release the next helm with this snapshot repository.
> 
> This PR is never urgent, so, if you feel the BanyanDB helm repo is ready to release, please go ahead to do so. I believe no matter banyandb helm repo is relying on BanyanDB server 0.5 release or not, it is close to be ready.

Let's wait for banyandb's first official release then.

Comment (hanahmily) [reply chart/skywalking/Chart.yaml:43]: Agreed. @ButterBright, could you please release the banyandb helm chart version 0.1.0? @kezhenxu94, would you kindly guide them through the process?

Comment (kezhenxu94) [reply chart/skywalking/Chart.yaml:43]: > Agreed. @ButterBright, could you please release the banyandb helm chart version 0.1.0? @kezhenxu94, would you kindly guide them through the process?

Reach out to me if there is any help I can offer, btw, it's too troublesome to ask a non-committer to release, even it's a committer who is releasing, one of the steps in release process still needs a PMC member to help, so basically there is few stuff can be done by @ButterBright to propose a release yet.

Comment (wu-sheng) [reply chart/skywalking/Chart.yaml:43]: We can't accept non committer to do a release. Because ASF requires apache ID and his/her sign based on that.
@hanahmily You need to take the responsibility to be release manager.

File: chart/skywalking/templates/_helpers.tpl
Comment (innerpeacez) [inline chart/skywalking/templates/_helpers.tpl:127]:
```suggestion

```

File: test/e2e/setup-e2e-shell/install.sh
Comment (innerpeacez) [inline test/e2e/setup-e2e-shell/install.sh:30]:
```suggestion
mkdir -p $TMP_DIR && cd $TMP_DIR
```

File: test/e2e/setup-e2e-shell/install-helm.sh
Comment (innerpeacez) [inline test/e2e/setup-e2e-shell/install-helm.sh:28]: The `|` pipeline seems to be unnecessary.

File: chart/skywalking/Chart.yaml
Comment (hanahmily) [reply chart/skywalking/Chart.yaml:43]: I will take care of the release

**Review by wu-sheng — APPROVED** [review]
Unblock

**Comment by wu-sheng** [comment]
@innerpeacez Please recheck.

File: README.md
Comment (innerpeacez) [inline README.md:73]:
```suggestion
git clone https://github.com/apache/skywalking-helm
```

File: README.md
Comment (innerpeacez) [inline README.md:74]:
```suggestion
cd skywalking-helm
```

File: .github/workflows/e2e.ci.yaml
Comment (wu-sheng) [inline .github/workflows/e2e.ci.yaml:74]: We should run e2e in the parallel mode. You could ref the main repo skywalking.yml to use `matrix` to define `name` and `config`.

**Review by wu-sheng — CHANGES_REQUESTED** [review]

**Comment by wu-sheng** [comment]
You need to resolve the conflicts, maybe your local branch is not up-to-date?

**Comment by ButterBright** [comment]
> You need to resolve the conflicts, maybe your local branch is not up-to-date?

Oh yeah, I see.

**Comment by wu-sheng** [comment]
Please check the task names, they have the same recent suffix. They should have meaningful suffix, such as banyandb, swck...

**Comment by ButterBright** [comment]
> Please check the task names, they have the same recent suffix. They should have meaningful suffix, such as banyandb, swck...

Can I simply make it the same as e2e test name? For example, Run Skywalking E2E Test (Elasticsearch as database), which I think also makes sense.

**Review by wu-sheng — APPROVED** [review]

**Comment by wu-sheng** [comment]
> > Please check the task names, they have the same recent suffix. They should have meaningful suffix, such as banyandb, swck...
> 
> Can I simply make it the same as e2e test name? For example, Run Skywalking E2E Test (Elasticsearch as database), which I think also makes sense.

Oops, yes, please do that. I accidentally merge this. Please open another PR to rename. @ButterBright

**Pull request merged**
